-1 
Tested at: 0f00168b62ce4b53a235a5a1a1bf3e1586c0246d
 I found an error when building: 
 
 <pre>>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/EgammaHLTProducers/src/EgammaHLTHybridClusterProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/EgammaHLTProducers/src/EgammaHLTIslandClusterProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/EgammaHLTProducers/src/EgammaHLTMulti5x5ClusterProducer.cc 
>> Compiling edm plugin /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/EgammaHLTProducers/src/EgammaHLTNxNClusterProducer.cc 
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/ElectronIdentification/src/ElectronLikelihood.cc: In member function 'void ElectronLikelihood::getInputVar(const reco::GsfElectron&, std::vector<float>&, const EcalClusterLazyTools&) const':
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/ElectronIdentification/src/ElectronLikelihood.cc:326:40: error: use of deleted function 'EcalClusterLazyToolsT<EcalClusterToolsT<false> >::EcalClusterLazyToolsT(const EcalClusterLazyToolsT<EcalClusterToolsT<false> >&)'
   EcalClusterLazyTools myEcalCluster = _myEcalCluster;
                                        ^
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/ElectronIdentification/interface/ElectronIDAlgo.h:14:0,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/ElectronIdentification/interface/ElectronLikelihood.h:7,
                 from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_6_X_2016-03-02-1100/src/RecoEgamma/ElectronIdentification/src/ElectronLikelihood.cc:6:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13497/11672/summary.html

--- a/RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h
+++ b/RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h
@@ -78,7 +78,7 @@ class EcalClusterLazyToolsBase {
   
   edm::EDGetTokenT<EcalRecHitCollection> ebRHToken_, eeRHToken_, esRHToken_;
 
-  std::unique_ptr<CaloSubdetectorTopology> ecalPS_topology_;
+  std::shared_ptr<CaloSubdetectorTopology> ecalPS_topology_;
 
   //const EcalIntercalibConstantMap& icalMap;
   edm::ESHandle<EcalIntercalibConstants> ical;

--- a/RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h
+++ b/RecoEcal/EgammaCoreTools/interface/EcalClusterLazyTools.h
@@ -50,7 +50,7 @@ class EcalClusterLazyToolsBase {
   // mapping for preshower rechits
   std::map<DetId, EcalRecHit> rechits_map_;
   // get Preshower hit array
-  std::vector<float> getESHits(double X, double Y, double Z, const std::map<DetId, EcalRecHit>& rechits_map, const CaloGeometry* geometry, CaloSubdetectorTopology *topology_p, int row=0, int plane=1);
+  std::vector<float> getESHits(double X, double Y, double Z, const std::map<DetId, EcalRecHit>& rechits_map, const CaloGeometry* geometry, CaloSubdetectorTopology const *topology_p, int row=0, int plane=1);
   // get Preshower hit shape
   float getESShape(const std::vector<float>& ESHits0);
   // get Preshower effective sigmaRR
@@ -78,7 +78,7 @@ class EcalClusterLazyToolsBase {
   
   edm::EDGetTokenT<EcalRecHitCollection> ebRHToken_, eeRHToken_, esRHToken_;
 
-  std::shared_ptr<CaloSubdetectorTopology> ecalPS_topology_;
+  std::shared_ptr<CaloSubdetectorTopology const> ecalPS_topology_;
 
   //const EcalIntercalibConstantMap& icalMap;
   edm::ESHandle<EcalIntercalibConstants> ical;

--- a/RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc
+++ b/RecoEcal/EgammaCoreTools/src/EcalClusterLazyTools.cc
@@ -301,7 +301,7 @@ float EcalClusterLazyToolsBase::eseffsiyiy(const reco::SuperCluster &cluster)
 }
 
 // get Preshower Rechits
-std::vector<float> EcalClusterLazyToolsBase::getESHits(double X, double Y, double Z, const std::map<DetId, EcalRecHit>& _rechits_map, const CaloGeometry* geometry, CaloSubdetectorTopology *topology_p, int row, int plane) 
+std::vector<float> EcalClusterLazyToolsBase::getESHits(double X, double Y, double Z, const std::map<DetId, EcalRecHit>& _rechits_map, const CaloGeometry* geometry, CaloSubdetectorTopology const *topology_p, int row, int plane) 
 {
   std::map<DetId, EcalRecHit> rechits_map = _rechits_map;
   std::vector<float> esHits;

